app/assets/v2/js/grants/fund.js

```diff
@@ -104,19 +133,38 @@ $(document).ready(function() {
               deployedSubscription.methods.getSubscriptionHash(...parts).call(function(err, subscriptionHash) {
                 $('#subscription_hash').val(subscriptionHash);
                 web3.eth.personal.sign('' + subscriptionHash, accounts[0], function(err, signature) {
-                  $('#signature').val(signature);
+                  if (signature) {
+                    $('#signature').val(signature);
+
+                    let data = {
+                      'subscription_hash': subscriptionHash,
+                      'signature': signature,
+                      'csrfmiddlewaretoken': $("#js-fundGrant input[name='csrfmiddlewaretoken']").val(),
+                      'sub_new_approve_tx_id': $('#sub_new_approve_tx_id').val()
+                    };
+
+                    $.ajax({
+                      type: 'post',
+                      url: '',
+                      data: data,
+                      success: function(json) {
+                        console.log('successfully saved subscriptionHash and signature');
+                        url = json.url;
+                        $('#wait').val(false);
+                      },
+                      error: function() {
+                        _alert({ message: gettext('Your subscription failed to save. Please try again.') }, 'error');
+                      }
+                    });
+                  }
                 });
               });
             });
           }).on('confirmation', function(confirmationNumber, receipt) {
-            $('#real_period_seconds').val(realPeriodSeconds);
 
             waitforData(function() {
-              $.each($(form).serializeArray(), function() {
-                data[this.name] = this.value;
-              });
               document.suppress_loading_leave_code = true;
-              form.submit();
+              window.location = url;
```

Comment:
This would result in `/undefined` added at the end of the url cause it's not assigned a value right ?

Reply:
+1

Reply:
I will go with solution number one